content/articles/spring-boot-caching/index.md

```diff
@@ -0,0 +1,332 @@
+        <expiry>
+            <ttl unit="minutes">5</ttl>
+        </expiry>
+        <resources>
+            <heap>1</heap>
+            <offheap unit="MB">1</offheap>
+        </resources>
+    </cache-template>
+</config>
+```
+- In the above code snippet, we set the time taken for the cache contents to be updated and the amount of memory to be used by the cache.
+		
+
+We need to inform our application of the path to the Ehcache configuration file. In the `application.properties` file in the resources directory, add the code snippet below.
+		
+
+```yaml
+spring.cache.jcache.config=classpath:ehcache.xml
+		
+```
+		
+
+#### Adding data into the cache
+In the `TodoServiceImpl.java` file, update the `getAllTodos()` method with the code snippet below.
+		
+
+```java
+@Cacheable(cacheNames = "todos")
+@Override
+public List<Todo> getAllTodos() {
+	return todoRepository.findAll();
+}
+```
+The above method returns a list of `todos` but is annotated with ` @Cacheable(cacheNames = "todos")` annotation that grabs the data returned from this method and stores them cache with a key `todos`.
+		
+
+> Data in the cache is stored in a key-value pattern. The key is the name stored in the cacheName variable while the value is the data that is returned by the method annotated with @Cacheable annotation.
+		
+To add a single item into the cache, we need to add the `id` that will be used to retrieve the item and updated the item in the cache.
+		
+
+Update the `getTodoById()` method with the code snippet below.
+```java
+@Cacheable(cacheNames = "todo", key = "#id")
+@Override
+public Todo getTodoById(Long id) {
+  return todoRepository.getById(id);
+}
+```
+		
+
+#### Updating data in the cache
+In the `TodoServiceImpl.java` file, update the `updateTodo()` method with the code snippet below.
+```java
+		
+@CachePut(value = "todos", key = "#todo.id")
+@Override
+public Todo updateTodo(Todo todo, Long id) {
+	 Todo newTodo = Todo
+	 .builder()
+	 .id(id)
+	 .description(todo.getDescription())
+	 .title(todo.getTitle())
+	 .build();
+	 todoRepository.save(newTodo);
+	 return newTodo;
+}
+```
+To update data in the cache, we use the ` @CachePut` annotation and provide the item's key to update.
+		
+
+#### Deleting data from the cache
+In the `TodoServiceImpl.java` file, update the `deleteTodo()` with the code snippet below.
+```java
+ @CacheEvict(value = "todo", key = "#id")
+ @Override
+ public void deletedTodo(Long id) {
+	 todoRepository.deleteById(id);
+}
+```
+To delete data from the cache, we use the `@CacheEvict` annotation passing in the key of the specific item to delete.
+		
+
+> To delete all items from the cache, we set the attribute `allEntries` to true.
+		
+### Conclusion
+Now that you have learned how to add caching to a Spring Boot application implement caching in your applications to reduce unnecessary database queries. You can download the complete source code (here)[https://replit.com/@elizabeth962/springbootcache#].
```

Comment:
This had an abrupt ending.

What have achieved by adding a cache to existing CRUD application? How is it different from the earlier?

More insights on this can be given in the conclusion.